src/testsets.jl

```diff
@@ -2,10 +2,10 @@
 mutable struct ReportingTestSet <: AbstractTestSet
     description::AbstractString
     results::Vector
+    properties::Dict{AbstractString, AbstractString}
```

Comment:
Do we really need this to be an AbstractSting?
Probably just a `String` is enough. All subtypes of `AbstractString` can be converted to a `String`.
and better performance when accessing it if not abstract.
Possibly should do same for all fields, though it doesn't really matter.
```suggestion
    properties::Dict{String, String}
```

Reply:
Probably not no, I can't imagine it would be used with anything but a `String`. For now I'll just change the ones on this PR, and put in another issue to review the others